rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -221,35 +235,39 @@ object SystemProcesses {
         phloRate  <- shortLeashParams.phloRate.get
         userId    <- shortLeashParams.userId.get
         timestamp <- shortLeashParams.timestamp.get
-        _ <- space
+        _ <- (space
               .produce(
                 ack,
                 ListParWithRandom(Seq(codeHash, phloRate, userId, timestamp), rand),
                 false
-              )(MATCH_UNLIMITED_PHLOS)
-              .map(unpackOption(_))
-              .foldResult(dispatcher)
+              )(MATCH_UNLIMITED_PHLOS))
+              .flatMap {
```

Comment:
Here you're already inside a for comprehension, so you really don't need the explicit `flatMap`